File: src/main/java/com/jetbrains/pluginverifier/verifiers/clazz/ClassVerifier.java
package com.jetbrains.pluginverifier.verifiers.clazz;

import com.jetbrains.pluginverifier.ErrorRegister;
import com.jetbrains.pluginverifier.resolvers.Resolver;
import org.objectweb.asm.tree.ClassNode;

public interface ClassVerifier {
  void verify(ClassNode clazz, Resolver resolver, ErrorRegister register);
}

File: intellij-plugin-verifier/verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/clazz/ClassVerifier.kt
/*
 * Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 */

package com.jetbrains.pluginverifier.verifiers.clazz

import com.jetbrains.pluginverifier.verifiers.VerificationContext
import com.jetbrains.pluginverifier.verifiers.resolution.ClassFile

interface ClassVerifier {
  fun verify(classFile: ClassFile, context: VerificationContext)
}
